### ISSUE TITLE
Variant Part cannot use chained LCS reference as with Linked Part 

### ISSUE THREAD

**Issue by OrdoScintilla** [Zolko-123/FreeCAD_Assembly4#505]
I have created an assembly with 2 LCSs on a first file.

On a separate file, I have an assembly, to reference multiple instances of my file.

When I use the function "Insert Part" which creates a Linked Part, or when I use the function "Create a Variant Part" the procedure is practically the same. Both work well for my part.

Yet, in the dialog box "Place Linked Part" when I insert a Linked Part, in the combo box "Attach to" the options include "Parent Assembly" as well as any other Linked Parts that have been inserted before.

![image](https://github.com/Zolko-123/FreeCAD_Assembly4/assets/66261803/d5e1aaac-455c-4fcc-9d74-dc01df7a15b6)

So I can insert a Link1 with LCS1 and LCS2. Then insert Link2 (same source part) and place a constraint to match LCS1 of Link1 with LCS2 of Link2, and so on. Thus, creating a chained reference. 

This is not the same case when using a Variant. When inserting a second variant, the first one doesn't appear in the combo box "Place Linked Part".

In case of first inserting a Link and then a Variant, the variant CAN be chained to the Link but not viceversa.

Given that the original part in the first file is the same, I don't understand why this is. I've tried creating assemblies where the LCS is subortinated to a Part inside an Assembly and Assembly with and without Part. What can be the explanation? Is this a problem with me knowing the right building procedure or is it a programming issue?

Thanks

**Comment by Zolko-123** [comment]
> What can be the explanation ? 

yes this is the expected behavior. The explanation is the following: a normal _Link_ is a link to an object in a document (file) on the hard-disk. For a _variantLink_ , the object is first copied into a new hidden temporary document, then modified as per variant variables, and then linked into the assembly. This means that a _variantLink_ is a normal link to a temporary object that is re-built at each re-opening, therefore the document is different at each session. Thus, references to this temporary object would change at each re-openening, and that defies the current logic of Asm4.

**Comment by OrdoScintilla** [comment]
Thanks,
The way I see it, this is a needed feature at this moment for me with the Variant. Because every positioning I have to do it's necessary for me to build the positioning manually with my own variables, and need to remember exactly what is where.
Since the positioning screen only allows numerical values and 90 degrees angle variations, I do it manually in the "data" tab for the variant, inside "assembly" group. This works well, and persists, the hassle is that it is all positioned from the assembly's origin and I have to calculate my variables and subtract or add the width of my profile because the variant's sketch constraints are not translated into the LCS's I created, as in the Linked Part.
Maybe this could be

**Comment by Zolko-123** [comment]
> for me to build the positioning manually with my own variables

think again : the assembly knows the variables of the _variantLink_ – since it's the assembly that assigns them – therefore you should be able to calculate the position of the target _LCS_ in the _variantLink_ directly from within the assembly. For example if the _variantLink_ is for a custom length of a an extruded beam, you can create an _LCS_ in the assembly, offset by the same value as the variable of the _variantLink_. 

And from the coding point of view it's far from trivial, the current behavior is a best compromise
